crates/holochain/src/conductor/conductor.rs

```diff
@@ -1905,6 +1933,20 @@ mod clone_cell_impls {
                 ));
             }
 
+            if let Some(dpki) = self.running_services().dpki {
+                let state = self.get_state().await?;
+                let app = state.get_app(installed_app_id)?;
+                let agent_key = app.agent_key().clone();
+                let key_state = dpki
+                    .state()
+                    .await
+                    .key_state(agent_key.clone(), Timestamp::now())
+                    .await?;
+                if let KeyState::Invalid(_) = key_state {
+                    return Err(AppError::CellToCloneHasInvalidAgent(agent_key).into());
+                }
+            }
```

Comment:
I think that rather, since we're looking at our own chain here, you should be looking at the chain itself. Even without DPKI, someone could choose to delete their agent key, thus invalidating their chain.